component/main.jsonnet

```
@@ -5,6 +5,144 @@ local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.topolvm;
 
+local namespace = kube.Namespace(params.namespace) {
+  metadata+: {
+    annotations+: {
+      'openshift.io/node-selector': '',
+    },
+    labels+: {
+      'app.kubernetes.io/name': params.namespace,
+      'topolvm.cybozu.com/webhook': 'ignore',
+    },
+  },
+};
+
+local lvmd_configmap = kube.ConfigMap('topolvm-lvmd0') {
+  metadata+: {
+    labels+: {
+      idx: '0',
+      'app.kubernetes.io/name': 'topolvm',
+      'app.kubernetes.io/instance': 'topolvm',
+      'app.kubernetes.io/version': '0.8.3',
+    },
+  },
+  data: {
+    'lvmd.yaml': |||
+      socket-name: /run/topolvm/lvmd.sock
+      device-classes:
+      - name: %(name)s
+        default: %(default)s
+        spare-gb: %(sparegb)d
+        volume-group: %(volumegroup)s
+    ||| % {
+      name: params.deviceclasses[0].name,
+      default: 'true',
+      sparegb: params.deviceclasses[0].sparegb,
+      volumegroup: params.deviceclasses[0].volumegroup,
+    },
+  },
+};
```

Use the new parameter format and use `std.manifestYamlDoc` instead of string interpolation to generate the lvmd config:

```suggestion
local lvmd_config = {
  'socket-name': '/run/topolvm/lvmd.sock',
  'device-classes': [
    params.deviceclasses[name] + { name: name }
    for name in std.objectFields(params.deviceclasses)
  ]
};
local lvmd_configmap = kube.ConfigMap('topolvm-lvmd0') {
  metadata+: {
    labels+: {
      idx: '0',
      'app.kubernetes.io/name': 'topolvm',
      'app.kubernetes.io/instance': 'topolvm',
      'app.kubernetes.io/version': '0.8.3',
    },
  },
  data: {
    'lvmd.yaml': std.manifestYamlDoc(lvmd_config),
  },
};
```

---

Fantastic, that would also allow me to easaly implement multiple device classes!